src/components/CIPipelineN/CIPipeline.tsx

```diff
@@ -191,18 +199,31 @@ export default function CIPipeline({
     })
     const validationRules = new ValidationRules()
     const [mandatoryPluginData, setMandatoryPluginData] = useState<MandatoryPluginDataType>(null)
+    const [pluginDataStore, setPluginDataStore] = useState<PluginDataStoreType>(
+        structuredClone(DEFAULT_PLUGIN_DATA_STORE),
+    )
+    const [availableTags, setAvailableTags] = useState<string[]>([])
```

Comment:
@vivek-devtron Pls suggest if these states should be moved to the provider here since the same are being reused

Reply:
Here currently we are using existing context with no provider, this should be picked in a refactoring story, for now would move the states related to this story to a provider and feed other states as props to that provider